Error when trying to use /swa immunity

Whenever I try to do /swa immunity, it gives off an error, something like: 
`ERROR null
30.03 03:56:24 [Server] INFO org.bukkit.command.CommandException: Unhandled exception executing command 'swa' in plugin SiegeWar v0.3.4
30.03 03:56:24 [Server] INFO at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchCommand(CraftServer.java:810) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerConnection.handleCommand(PlayerConnection.java:2093) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerConnection.c(PlayerConnection.java:1908) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1861) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:43) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1199) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1192) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:1168) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1012) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:178) ~[patched_1.16.5.jar:git-Tuinity-"44e8e5d"]
30.03 03:56:24 [Server] INFO at java.lang.Thread.run(Thread.java:748) [?:1.8.0_211]
30.03 03:56:24 [Server] INFO Caused by: java.lang.ArrayIndexOutOfBoundsException: 1`

**To Reproduce**
Steps to reproduce the behavior:
1. Do /swa immunity

**Expected behavior**
The command should work

**Screenshots**
![Screenshot from 2021-03-30 10-03-14](https://user-images.githubusercontent.com/54766911/112932119-313d4c00-913f-11eb-95d3-01a48485e8d3.png)


**Versions (please complete the following information):**
 - Towny: 0.96.7.10
 - SiegeWar: 0.3.4
 - Spigot/Paper:This server is running Tuinity version git-Tuinity-"44e8e5d" (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)

**Files**
Startup log: 
[latest (3).log](https://github.com/TownyAdvanced/SiegeWar/files/6226101/latest.3.log)

SiegeWar config:
[config.zip](https://github.com/TownyAdvanced/SiegeWar/files/6226106/config.zip)



Townyperms.yml:
[townyperms.zip](https://github.com/TownyAdvanced/SiegeWar/files/6226107/townyperms.zip)

--- a/src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java
+++ b/src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java
@@ -203,18 +203,23 @@ public class SiegeWarAdminCommand implements CommandExecutor, TabCompleter {
 	}
 
 	private void parseSiegeWarImmunityCommand(CommandSender sender, String[] args) {
+		if (args.length < 2) {
+			showImmunityHelp(sender);
+			return;
+		}
+
 		try {
 			if (args[0].equalsIgnoreCase("alltowns"))
 				Integer.parseInt(args[1]);
 			else
 				Integer.parseInt(args[2]);
-		} catch (NumberFormatException e) {
+		} catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
 			Messaging.sendMsg(sender, Translation.of("msg_error_must_be_num"));
 			showImmunityHelp(sender);
 			return;
 		}
 
-		if (args.length == 3 && args[0].equalsIgnoreCase("town")) {
+		if (args.length >= 3 && args[0].equalsIgnoreCase("town")) {
 			//town {townname} {hours}
 			Town town = TownyUniverse.getInstance().getTown(args[1]);
 			if (town == null) {
@@ -226,7 +231,7 @@ public class SiegeWarAdminCommand implements CommandExecutor, TabCompleter {
 			TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
 			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
 
-		} else if (args.length == 3 && args[0].equalsIgnoreCase("nation")) {
+		} else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
 			//nation {nationname} {hours}
 			Nation nation = TownyUniverse.getInstance().getNation(args[1]);
 			if (nation == null) {
@@ -240,8 +245,7 @@ public class SiegeWarAdminCommand implements CommandExecutor, TabCompleter {
 			TownyMessaging.sendPrefixedNationMessage(nation, Translation.of("msg_set_siege_immunities_nation", args[1], args[2]));
 			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_nation", args[1], args[2]));
 
-		} else if(args.length == 2
-			&& args[0].equalsIgnoreCase("alltowns")) {
+		} else if(args.length >= 2 && args[0].equalsIgnoreCase("alltowns")) {
 			//all towns
 			long durationMillis = (long)(Long.parseLong(args[1]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
 			for (Town town : new ArrayList<>(TownyUniverse.getInstance().getTowns()))  {

